pkg/convert/apkbuild/apkbuild.go

```diff
@@ -345,33 +349,45 @@ func (c ApkConvertor) mapconvert() {
-	//	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "meson/compile"})
-	//	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "meson/install"})
-	//
-	// case BuilderTypeMake:
-	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/configure"})
-	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/make"})
-	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/make-install"})
-	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "strip"})
-
-	//default:
-	//	c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "# FIXME"})
-	//
-	//}
+	if c.Apkbuild.Funcs["build"] != nil {
+		var output bytes.Buffer
+		printer := syntax.NewPrinter()
+		err := printer.Print(&output, c.Apkbuild.Funcs["build"])
+		if err != nil {
+			fmt.Errorf("failed to print build function: %w", err)
+		}
+
+		outputStr := output.String()
+
+		// Define the regular expression to find options in the form of -<option>=<value> or --<option>=<value>
+		re := regexp.MustCompile(`--?\w[-\w]*=[^\\\s]+`)
+
+		// Find all matches
+		matches := re.FindAllString(outputStr, -1)
+
+		// Join matches into a single string separated by space and backslash
+		opts := strings.Join(matches, " \\\n")
+
+		switch {
+		case strings.Contains(outputStr, "cmake"):
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "cmake/configure"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "cmake/build"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "cmake/install"})
+		case strings.Contains(outputStr, "meson"):
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "meson/configure", With: map[string]string{
+				"opts": opts,
+			}})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "meson/compile"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "meson/install"})
+		case strings.Contains(outputStr, "autoconf"):
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/configure"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/make"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "autoconf/make-install"})
+			c.GeneratedMelangeConfig.Pipeline = append(c.GeneratedMelangeConfig.Pipeline, config.Pipeline{Uses: "strip"})
+			// default:
```

Comment:
We might want to keep the default as "make" as before, or at least scaffold out the metadata with no pipeline?